txpipe/rail_pz.py

```diff
@@ -0,0 +1,190 @@
+from .base_stage import PipelineStage
+from .data_types import PhotozPDFFile, HDFFile, PickleFile
+import numpy as np
+
+
+class PZRailTrain(PipelineStage):
+    """Train a photo-z model using RAIL.
+
+    The Redshift Assessment Infrastructure Layers (RAIL) library provides a uniform
+    interface to DESC photo-z code.
+
+    TXPipe uses RAIL across several different pipeline stages.
+    This stage, which would normally be run first, uses a training set (e.g. of
+    spectroscopic redshifts) to train and save an Estimator object that a later
+    stage can use to measure redshifts of the survey sample.
+    """
+    name = "PZRailTrain"
+
+    inputs = [
+        ("photoz_training", HDFFile),
+        ("photoz_testing", HDFFile),
+    ]
+
+    outputs = [("photoz_trained_model", PickleFile)]
+
+    config_options = {
+        "class_name": str,
+        "chunk_rows": 10000,
+        "zmin": 0.0,
+        "zmax": 3.0,
+        "nzbins": 301,
+        # other estimator-specific config
+        # can be put in the config file and
+        # will end up in self.config
+    }
+
+    def run(self):
+        from rail.estimation.estimator import Estimator
+
+        # General config information that RAIL wants
+        base_config = {
+            "trainfile": self.get_input("photoz_training"),
+            "testfile": self.get_input("photoz_testing"),
+            "hdf5_groupname": "photometry",
+            "chunk_size": self.config["chunk_rows"],
+            "outpath": None,  # should not be used here
+        }
+
+        # Additional confguration specific to this algorithm
+        # can be included in the config
+        run_dict = {
+            "run_params": self.config.copy(),
+        }
+
+        # create an instance of the specific estimation
+        # method
+        cls = Estimator._find_subclass(self.config["class_name"])
```

Comment:
looks like the underscore is not compatible with such a use, and not warranted, as this usage seems reasonable. I wonder if we can/should add classmethod __call__ to Estimator, that would hide the exact same call.

Reply:
I've asked about this here: https://github.com/LSSTDESC/RAIL/issues/45#issuecomment-756201258